contractcourt/channel_arbitrator.go

```diff
@@ -656,6 +656,54 @@ func (c *ChannelArbitrator) relaunchResolvers(commitSet *CommitSet,
 			resolver.SupplementState(chanState)
 		}
 
+		// The on disk resolutions contains all the ctrl block
+		// information, so we'll set that now for the relevent
+		// resolvers.
+		switch r := resolver.(type) {
```

Comment:
This is only needed for taproot channels and we could make it into a new method.

Reply:
Hmm, I think the only way to detect that with the given data structures would be to peer several layers deep down into the control block for the local/remote outputs. Will see if there's another easier path...